### DISCUSSION TITLE
Would it be possible for KeePassXC to store gpg keys and add them to gpg-agent like it does with ssh keys?

### DISCUSSION THREAD

**Discussion by PoisonFrog** [keepassxreboot/keepassxc#9278] [Q&A] [answered]
I love the fact that you can store SSH keys in your KeePass database and have them added to the ssh-agent automatically when you open your database or on demand, but could a feature that does the same to gpg keys be added to KeePassXC? Would it be even possible? Like storing private and public keys on KeePass and adding them to the agent once you open the database?

**Comment by droidmonkey** [comment, marked as answer]
In general we are not interested in doing that. Certificate management is far more complicated than ssh-key management. There are already excellent tools for this, Kleopatra being the best.

**Reply by PoisonFrog** [reply]
Understandable, but I don't want it to replace gnupg/kleopatra. I wanted to make these certificates portable without having to worry about importing the ~/.gnupg directory from somewhere else. It would be all inside the keepass database. You would still use gnupg or kleopatra to interact with the keys. KeePassXC would just add them to the agent. I just thought it would be super convenient, but if it would add a lot of complexity to the application or if it is too hard to implement then it is probably not worth it.

Thanks for the answer.

**Reply by droidmonkey** [reply]
You can certainly store the certs as attachments to an entry and save them to disk when you want to import into gpg/kleo. Integration with those tools at this point is not on the radar.

**Reply by pryndor** [reply]
If I attach a TXT file within KeePassXC, can I still access the same file later even if I delete it locally, or do I need to keep the same file in the local folder as well? Apart from TXT files, what other file formats can be stored in KeePassXC?

**Reply by the-wolfman** [reply]
@Bala-periannan Attachments are not limited to a specific file format. You can - in theory - put in there whatever you like. But keep in mind your use case.  I don't recommend using it as an "encrypted disc" replacement. There are other encrypted containers more suited. That's why:
- big files or a big number of files will bloat your single file database and may make it hard to handle
- KeepassXC does not provide a file-system emulation. You can import files into the database, export files from the database, but you can not access the files in the database like they would be sitting in your filessystem. There is no path to them nor can you link to them.

**Comment by Chealer** [comment]
There could be a plugin to do that, as ticket #4048 requests. Meanwhile, as explained there, you could:
1. Create a pseudo-entry just to store each OpenPGP key’s password
2. Setup [Auto-Type](https://github.com/keepassxreboot/keepassxc/blob/develop/docs/topics/AutoType.adoc), if you want to enter the password faster.
